ocaml/xapi/authx.ml

```diff
@@ -134,7 +134,11 @@ module AuthX : Auth_signature.AUTH_MODULE = struct
         (* no exception raised, then authentication succeeded *)
       with Failure msg -> raise (Auth_signature.Auth_failure msg)
     in
-    get_subject_identifier username
+    try get_subject_identifier username
+    with Not_found ->
+      raise
+        (Auth_signature.Auth_failure
+           (Printf.sprintf "Could not find user's SID (uname='%s')" username))
```

Comment:
The actual reason is that neither the user's id nor the group id could be found.